### DISCUSSION TITLE
Use MJML for email templates

### DISCUSSION THREAD

**Discussion by eikaramba** [directus/directus#10809] [Feature Requests]
This is a follow up ticket to https://github.com/directus/directus/issues/10460

### Idea
Use the framework [MJML](https://mjml.io/) for writing email templates instead of manually writing the email html. Manually writing good cross-client compatible emails is very hard even in 2022. MJML is a html-to-html transpiler helping in that.

### Incorporating into directus
MJML is not a template engine. Thus an existing engine still needs to be used for e.g. conditional statements or loops and variables. Directus is currently using liquid.js - either that or any other engine. I have used https://github.com/forwardemail/email-templates in the past which basically let's the user decide which engine to use. this is not needed here, just another possible change.

After rendering the html with liquid.js it would be needed to invoke the mjml compiler. e.g. in https://github.com/directus/directus/blob/main/api/src/services/mail/index.ts#L55
```js
const mjml = require("mjml");
....
html = await this.renderTemplate(template.name, templateData);
let mjmloutput = mjml(html, {});
```

### caveats
Not everyone wants or can use MJML. Therefore it would be best to let the user decide and maybe follow a naming schema like mytemplate.mjml.liquid. Another approach which is nice is what email-templates is doing. here a folder-filename structure is used like
`mytemplate/[text|html].enginename` -> for example `invoices/html.liquid`. one could adapt this and use mjml.liquid as mjml is only outputting html.

Before implementing anything i would like to know whether this is something directus is interested in incorporating into the core framework. There might be also a way to just use a custom js class which accepts the email service as a parameter and wraps around it to provide the same functionality. I am fine with that too, just want to know whether i should start working on this as i need this functionality myself.

**Comment by rijkvanzanten** [comment]
I'm wondering what we could do to keep the configuration minimal. I'd rather not add too many opinionated tools into the core platform without them being extendable (otherwise we'll have a "Support XYZ. You support MJML, why not XYZ instead?" soon enough). Does MJML ignore regular HTML? Could we simply always pass it through MJML on render? 🤔 

Also from a quick peak at their docs, it looks like MJML itself doesn't have any templating / variables support, so we'd still have to rely on something like liquid to inject the variables before passing it to MJML.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by readtedium** [reply]
If by chance this ever gets a revival, I will say that MJML is probably the best tool of its kind in the email space and one that I think would be beneficial for any email implementation. I personally use it a lot.
